elastic/apm-server
# Extrapolate Jaeger transaction count from reported sampling rate

## Motivation/summary

When Jaeger is configured to sample a percentage of traces, then the statistics reported in APM UI will be proportional to the sampling rate, and not the actual number of operations. Jaeger reports the sampling rate as a pair of tags (sampler.type and sampler.param) with each span. We can use these to extrapolate the number of transactions when performing aggregations.

## Checklist


## How to test these changes

1. Enable transaction aggregation (`apm-server.aggregation.enabled=true`)
2. Run an app instrumented with Jaeger, configured with remote sampling
3. Set a sampling rate of 0.1 for the service in APM UI
4. Run 1000 operations
5. Check there are 100 (i.e. 1000 * 0.1) transactions
6. Check the sum of counts in `transaction.duration.histogram` fields is 1000 (e.g. using https://github.com/elastic/elasticsearch/pull/55933 when it lands)

## Related issues

Closes #3011
